jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/URIUtil.java

```diff
@@ -1845,10 +1847,52 @@ public static URI toURI(String resource)
     {
         Objects.requireNonNull(resource);
 
-        // Only try URI for string for known schemes, otherwise assume it is a Path
-        return (ResourceFactory.isSupported(resource))
-            ? correctFileURI(URI.create(resource))
-            : Paths.get(resource).toUri();
+        if (URIUtil.hasScheme(resource))
+        {
+            try
+            {
+                URI uri = new URI(resource);
+
+                if (ResourceFactory.isSupported(uri))
+                    return correctFileURI(uri);
```

Comment:
I don't like the method name as it looks like it is assuming a file URI, plus we might have other corrections we need to do in future.  How about:

```suggestion
                    return correctedURI(uri);
```

Reply:
This would be a method signature change on a released method.
And yes, your assumption is correct, it ONLY does file (scheme) URI corrections.

Reply:
So rename and add the old method back as deprecated/marked-for-removal that calls the new name.

Reply:
Done.